{
  type=throwable.getClass().getName();
  try {
    stackTrace=throwable.getStackTrace();
  }
 catch (  Throwable ignored) {
    LOGGER.debug("Ignoring failure to extract throwable stack trace.",ignored);
    stackTrace=new StackTraceElement[0];
  }
  try {
    message=throwable.getMessage();
  }
 catch (  Throwable failure) {
    getMessageExec=failure;
  }
  try {
    toString=throwable.toString();
  }
 catch (  Throwable failure) {
    toStringRuntimeExec=failure;
  }
  Throwable causeTmp;
  try {
    causeTmp=throwable.getCause();
  }
 catch (  Throwable ignored) {
    LOGGER.debug("Ignoring failure to extract throwable cause.",ignored);
    causeTmp=null;
  }
  final Throwable causeFinal=causeTmp;
  ByteArrayOutputStream outstr=new ByteArrayOutputStream();
  ObjectOutputStream oos=new ExceptionReplacingObjectOutputStream(outstr){
    boolean seenFirst;
    @Override protected Object replaceObject(    Object obj) throws IOException {
      if (!seenFirst) {
        seenFirst=true;
        return obj;
      }
      if (obj == causeFinal) {
        return new CausePlaceholder();
      }
      return super.replaceObject(obj);
    }
  }
;
  try {
    oos.writeObject(throwable);
    oos.close();
    serializedException=outstr.toByteArray();
  }
 catch (  Throwable ignored) {
    LOGGER.debug("Ignoring failure to serialize throwable.",ignored);
  }
  if (causeFinal != null) {
    cause=new ExceptionPlaceholder(causeFinal);
  }
}
